### PR TITLE
Add windowsperfcounters receiver from contrib

### PR DESCRIPTION
**Pull request by chrislbs**

### CODE REVIEW

**Comment by dmitryax** [comment]
@chrislbs can you please rebase your PR?

**Comment by dmitryax** [comment]
> please wait for us to complete our internal mapping rules before merging

There shouldn't be any mapping rules required here. This receiver doesn't have any predefined set of metrics and has to be explicitly configured

File: docs/components.md [hunk 1]
```diff
@@ -36,22 +36,23 @@ support at this time). While in alpha, breaking changes may be introduced in a
...
+| [statsd](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/statsdreceiver)                             |            |                                                                                                     |            |
+| [syslog](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/syslogreceiver)                             |            |                                                                                                     |            |
+| [tcplog](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/tcplogreceiver)                             |            |                                                                                                     |            |
+| [windowsperfcounters](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowsperfcountersreceiver)   |            |                                                                                                     |            |
```
Comment (dmitryax) [inline docs/components.md:55]: nit: maybe the column is unnecessary longer than needed 
```suggestion
| [windowsperfcounters](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowsperfcountersreceiver) |            |                                                                                                     |            |
```

File: docs/components.md [hunk 2]
```diff
@@ -36,22 +36,23 @@ support at this time). While in alpha, breaking changes may be introduced in a
...
+| [sqlquery](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/sqlqueryreceiver)                         |            |                                                                                                     |            |
+| [statsd](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/statsdreceiver)                             |            |                                                                                                     |            |
+| [syslog](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/syslogreceiver)                             |            |                                                                                                     |            |
+| [tcplog](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/tcplogreceiver)                             |            |                                                                                                     |            |
```
[the remark this answers is not in the document]
Comment (dmitryax) [reply docs/components.md:54]: I believe the formatting changes are required because the table is extended due to long name of the new receiver

**Comment by chrislbs** [comment]
Thanks for the feedback. I rebased to the latest off of main.

File: docs/components.md (hunk 1 above)
Comment (chrislbs) [reply docs/components.md]: I fixed this during the rebase.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**